src/components/Modal.js

```diff
@@ -31,55 +32,87 @@ const defaultProps = {
     type: '',
 };
 
-const Modal = (props) => {
-    const {
-        modalStyle,
-        modalContainerStyle,
-        swipeDirection,
-        animationIn,
-        animationOut,
-        needsSafeAreaPadding,
-        hideBackdrop,
-    } = getModalStyles(props.type, useWindowDimensions());
+class Modal extends Component {
+    constructor(props) {
+        super(props);
 
-    return (
-        <ReactNativeModal
-            onBackdropPress={props.onClose}
-            onBackButtonPress={props.onClose}
-            onSwipeComplete={props.onClose}
-            swipeDirection={swipeDirection}
-            isVisible={props.isVisible}
-            backdropColor={themeColors.modalBackdrop}
-            backdropOpacity={hideBackdrop ? 0 : 0.5}
-            backdropTransitionOutTiming={0}
-            style={modalStyle}
-            animationIn={animationIn}
-            animationOut={animationOut}
-        >
-            <CustomStatusBar />
-            <SafeAreaInsetsContext.Consumer>
-                {(insets) => {
-                    const {paddingTop, paddingBottom} = getSafeAreaPadding(insets);
-                    return (
-                        <View
-                            style={{
-                                // This padding is based on the insets and could not neatly be
-                                // returned by getModalStyles to avoid passing this inline.
-                                paddingTop: needsSafeAreaPadding ? paddingTop : 20,
+        this.state = {
+            window: Dimensions.get('window'),
+        };
 
-                                ...styles.defaultModalContainer,
-                                paddingBottom,
-                                ...modalContainerStyle,
-                            }}
-                        >
-                            {props.children}
-                        </View>
-                    );
-                }}
-            </SafeAreaInsetsContext.Consumer>
-        </ReactNativeModal>
-    );
-};
+        this.onDimensionChange = this.onDimensionChange.bind(this);
+    }
+
+    componentDidMount() {
+        Dimensions.addEventListener('change', this.onDimensionChange);
```

Comment:
We're starting to add this logic in quite a few places now (it was previously in three places, now it's four). Is it time to move this into an HOC like `withWindowDimensions()`?

Reply:
Yep let's do it.